lib/ruby_lsp/ruby-lsp-shoulda-context/listeners/test_discovery.rb

```diff
@@ -0,0 +1,173 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RubyLsp
+  module ShouldaContext
+    class ShouldaTestStyle < Listeners::TestDiscovery
+      extend T::Sig
+      include Requests::Support::Common
+
+      sig { params(response_builder: ResponseBuilders::TestCollection, global_state: GlobalState, dispatcher: Prism::Dispatcher, uri: URI::Generic).void }
+      def initialize(response_builder, global_state, dispatcher, uri)
+        super
+        @response_builder = response_builder
+        @uri = uri
+        @visibility_stack = T.let([], T::Array[Symbol])
+        @nesting = T.let([], T::Array[String])
+        @context_block_nesting = T.let([], T::Array[String])
+
+        dispatcher.register(
+          self,
+          :on_class_node_enter,
+          :on_class_node_leave,
+          :on_call_node_enter,
+          :on_call_node_leave,
+        )
+      end
+
+      sig { params(node: Prism::ClassNode).void }
+      def on_class_node_enter(node)
+        @nesting << constant_name(node.constant_path)
+
+        # TODO: Improve based on ancestor classes
+        class_name = @nesting.join("::")
+        return unless class_name.end_with?("Test")
```

Comment:
This can be improved with the TestDiscover helpers to actually check the ancestors of the test class.